client/core/core.go

```
@@ -7046,3 +7047,140 @@ func (c *Core) DeleteArchivedRecords(olderThan *time.Time, matchesFile, ordersFi
+	makeError := func(err error) (*trackedTrade, []dex.Bytes, []dex.Bytes, dex.Bytes, uint64, error) {
+		return nil, nil, nil, nil, 0, err
+	}
+
+	if len(oidB) != order.OrderIDSize {
+		return makeError(fmt.Errorf("wrong order ID length. wanted %d, got %d", order.OrderIDSize, len(oidB)))
+	}
+	var oid order.OrderID
+	copy(oid[:], oidB)
+
+	for _, dc := range c.dexConnections() {
+		tracker, _, _ = dc.findOrder(oid)
+		if tracker != nil {
+			mktID := marketName(tracker.Base(), tracker.Quote())
+			mkt := dc.marketConfig(mktID)
+			if mkt == nil {
+				return makeError(fmt.Errorf("could not find market: %v", mktID))
+			}
+			lotSize := mkt.LotSize
+			fromAsset := dc.assets[tracker.fromAssetID]
+			if fromAsset == nil {
+				return makeError(fmt.Errorf("could not find asset with id: %v", tracker.fromAssetID))
+			}
+			swapSize := fromAsset.SwapSize
+			lotsRemaining := tracker.Trade().Remaining() / lotSize
+			requiredForRemainingSwaps = lotsRemaining * swapSize * tracker.metaData.MaxFeeRate
+			break
+		}
+	}
+	if tracker == nil {
+		return makeError(fmt.Errorf("could not find active order with id: %s", oid))
+	}
+
+	tracker.mtx.Lock()
+	defer tracker.mtx.Unlock()
+
+	swapCoins = make([]dex.Bytes, 0, len(tracker.matches))
+	for _, match := range tracker.matches {
+		if match.Status < order.MakerSwapCast {
+			continue
+		}
+		var swapCoinID order.CoinID
+		if match.Side == order.Maker {
+			swapCoinID = match.MetaData.Proof.MakerSwap
+		} else {
+			if match.Status < order.TakerSwapCast {
+				continue
+			}
+			swapCoinID = match.MetaData.Proof.TakerSwap
+		}
+		swapCoins = append(swapCoins, dex.Bytes(swapCoinID))
+	}
+
+	accelerationCoins = make([]dex.Bytes, 0, len(tracker.metaData.AccelerationCoins))
+	for _, coin := range tracker.metaData.AccelerationCoins {
+		accelerationCoins = append(accelerationCoins, dex.Bytes(coin))
+	}
+
+	if tracker.metaData.ChangeCoin == nil {
+		return makeError(fmt.Errorf("order does not have change which can be accelerated"))
+	}
```

You could move this check up below the lock and fail faster.